File: stock-market-service/src/main/java/rs/edu/raf/model/Stock.java
package rs.edu.raf.model;

import lombok.Getter;
import org.springframework.data.redis.core.RedisHash;

@RedisHash("Stock")
@Getter
public class Stock extends Listing {
    private Long outstandingShares;
    private Double dividendYield;
    private Double marketCap;

    public Stock(String ticker, String description, String exchange, Long lastRefresh, Double price, Double high,
                 Double low, Double change, Long volume, Long outstandingShares, Double dividendYield) {
        super(ticker, description, exchange, lastRefresh, price, high, low, change, volume, 1L, price / 2);
        this.outstandingShares = outstandingShares;
        this.dividendYield = dividendYield;
        this.marketCap = outstandingShares * price;
    }
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/model/Stock.kt
package org.example.stockmarketservicekotlin.model

import org.springframework.data.redis.core.RedisHash

@RedisHash("stock")
class Stock(
    ticker: String,
    description: String,
    exchange: String,
    lastRefresh: Long,
    price: Double,
    high: Double,
    low: Double,
    change: Double,
    volume: Long,
    contractSize: Long,
    maintenanceMargin: Double,
    changePercent: Double,
    dollarVolume: Double,
    nominalValue: Double,
    initialMarginCost: Double,
    val outstandingShares: Long,
    val dividendYield: Double,
    val marketCap: Double
) : Listing(
    ticker,
    description,
    exchange,
    lastRefresh,
    price,
    high,
    low,
    change,
    volume,
    contractSize,
    maintenanceMargin,
    changePercent,
    dollarVolume,
    nominalValue,
    initialMarginCost
) {
}